_posts/2020-12-01-infrastructure-migration-finished.md

```diff
@@ -0,0 +1,275 @@
+
+This was the longest migration I ever did: 11 months in the making.
+But we finally did it!
+
+This was mostly made possible by us receiving [AWS Open Source credits](https://aws.amazon.com/blogs/opensource/aws-promotional-credits-open-source-projects/).
+This allowed me to try out AWS services, do stupid things (rack up a bill of more than 500 dollar per month .. which is HUGE for us), and all while knowing it wouldn't clean out our donation-funds (read: the credits covered all our costs for 2020).
+Sadly, in 2021 we will not receive AWS Open Source credits .. this means no more experimenting.
+So a perfect time to write up what happened over the last 16 years, and more specifically, last 2 years.
+
+<!-- more -->
+
+First of all, this will be a long post.
+Just .. so you are prepared.
+
+We start with a bit of history:
+
+In 2004 openttd.org was a website, and a wiki, and that was about it.
+In the years after, we greatly extended what we offered on openttd.org:
+- Master server, so you have in-game server listing
+- Content services, so you can download NewGRFs, Scenarios, ... in-game
+- Bug-tracker
+- Mail
+- Maillists
+- IRC bots
+- Developers-space
+- Content Delivery Network (just called "mirrors" back in the days)
+- NoAI tournament servers
+- Centralized accounts
+- Subversion
+
+And I am sure that is not a complete list.
+We did weird stuff, not all ideas survived, but boy, we were busy.
+By 2007, most things were either invented or there was no interest in developing new services.
+
+(11 years later)
+
+In 2018, nothing much had changed on the infrastructure side.
+There were some minor additions, some tuning, but nothing really.
+BaNaNaS was still running the same software as it was in 2007.
+The centralized accounts still had "coming soon" on the profile page.
+
+Is this a bad thing, you ask?
+Well, yes and no.
+From a stability point of view, this was perfect.
+Our services could handle the weirdest things .. for example:
+- We once got [posted on slashdot](https://games.slashdot.org/story/10/04/06/0519206/OpenTTD-100-Released), which is called the slashdot-effect.
+- Once, a [popular YouTube channel](https://www.youtube.com/playlist?list=PLz1xmRxpXfWvEefxXiUo0C8stV6TAsKss) played OpenTTD, doubling our player-base overnight.
+- And many more (smaller) events.
+
+What this means is that many people come to your website, download the game.
+This sounds cool, but it also increases the load on the services.
+Often what you see in these cases, that things crack, and break.
+But .. not on openttd.org.
+The services survived such insane increase in traffic.
+To put it in numbers: we went from between 5 to 15 requests per second, to between 60 and 200 requests per seconds, depending on the event.
+This is an insane amount of additional traffic to handle.
+But, openttd.org could handle it.
+Well, there was this once: a user found a bug in our code, crashing the software .. during such event.
+That was fun :P
```

Comment:
"the software" could probably be replaced by master-server/website/whatever more specific
```suggestion
But, openttd.org could handle it.
Except for the time a user found a bug in our code that crashed the software during such event.
That was fun :P
```